pilot/pkg/model/telemetry.go

```diff
@@ -142,7 +142,7 @@ func shallowMergeTracing(parent, child *tpb.Telemetry) *tpb.Telemetry {
 	}
 
 	// TODO: use wrapper in API to allow 0-valued override
-	if childTracing.GetRandomSamplingPercentage() != 0 {
+	if childTracing.RandomSamplingPercentage.GetValue() != 0 {
```

Comment:
I don't think this is what we want, instead we should have 
```suggestion
	if childTracing.RandomSamplingPercentage != nil {
```

Your code keeps the same behavior as today, but we changed the API so we could change the same behavior

Reply:
Gotcha, fixed